src/core/lib/iomgr/fork_posix.c

```diff
@@ -47,36 +48,45 @@ int grpc_prefork() {
             "environment variable GRPC_ENABLE_FORK_SUPPORT=1");
     return 0;
   }
-  grpc_exec_ctx exec_ctx = GRPC_EXEC_CTX_INIT;
-  grpc_timer_manager_set_threading(false);
-  grpc_executor_set_threading(&exec_ctx, false);
-  grpc_exec_ctx_finish(&exec_ctx);
-  if (!gpr_await_threads(
-          gpr_time_add(gpr_now(GPR_CLOCK_REALTIME),
-                       gpr_time_from_seconds(3, GPR_TIMESPAN)))) {
-    gpr_log(GPR_ERROR, "gRPC thread still active! Cannot fork!");
-    return 0;
+  if (grpc_is_initialized()) {
+    grpc_exec_ctx exec_ctx = GRPC_EXEC_CTX_INIT;
+    grpc_timer_manager_set_threading(false);
+    grpc_executor_set_threading(&exec_ctx, false);
+    grpc_exec_ctx_finish(&exec_ctx);
+    if (!gpr_await_threads(
+            gpr_time_add(gpr_now(GPR_CLOCK_REALTIME),
+                         gpr_time_from_seconds(3, GPR_TIMESPAN)))) {
+      gpr_log(GPR_ERROR, "gRPC thread still active! Cannot fork!");
+      return 0;
```

Comment:
`grpc_prefork` doesn't need to return any pass/fail return codes, correct? Should this return void rather than the wrapper?

Reply:
Fixed